native/src/main/java/io/ballerina/stdlib/http/api/HttpDispatcher.java

```diff
@@ -444,7 +444,10 @@ private static void populatePathParams(Resource resource, Object[] paramFeed,
         for (String paramName : pathParamTokens) {
             String argumentValue = resourceArgumentValues.getMap().get(paramName).get(actualSignatureParamIndex);
             try {
-                argumentValue = URLDecoder.decode(argumentValue, "UTF-8");
+                if (argumentValue.endsWith("%")) {
```

Comment:
Can we use constants for these string characters?

Reply:
Fixed in [127f90a](https://github.com/ballerina-platform/module-ballerina-http/pull/1469/commits/127f90af09c2a8b857c665abefa735f7fa1c6a6b)